API.Tests/Services/ArchiveServiceTests.cs

```diff
@@ -291,7 +291,25 @@ public void CanParseComicInfo()
                 Web = "https://www.comixology.com/BTOOOM/digital-comic/450184"
             };
 
+            // XXX: This is not doing anything useful
             Assert.NotStrictEqual(expected, actual);
+            // XXX: This also works
+            Assert.NotStrictEqual(null, actual);
```

Comment:
Explain why this isn't doing anything useful, when we are validating what was parsed matches what we expect?

Reply:
First of all `NotStrictEqual` makes no sense here. If the prerequisite conditions were met, the test should have been:
```
Assert.StrictEqual(expected, actual);
```
It would have asserted that the objects types matched and the values matched instead of the opposite as it current stands.

The later check (value equality) relies on the class (here ComicInfo) implementing the Equals method, for checking equality of the "value". Because an explicit one is not defined for ComicInfo, the test here falls back to the one implemented on Object (https://docs.microsoft.com/en-us/dotnet/api/system.object.equals?view=net-6.0#system-object-equals(system-object)). The Object implementation is only true when the identifiers point to the same instance of an object and here `actual` and `expected` are 2 different instances of ComicInfo that happen to have the same values.

There are two ways to fix this test:
 1) implement an Equal() method on object ComicInfo and use `StrictEqual`.
 2) Check individually the field on the two objects like I did on the next test.
 
Please let me know how you'd rather see this addressed?

Reply:
I have added a partial implementation of `Equals()` to illustrate the issue with incorrect usage of NotStrictEqual.

On second though, implementing a correct `Equals()` method seems to be far too expensive to be justified by only a couple of unit tests. In particular, multiple values fields like Characters, etc would need something like being split and sorted before being compared.

Reply:
But we don't use any equality checks for ComicInfo anywhere in the code, so I don't think it would be that bad. Instead, we can rewrite the Unit test to be more like how you wrote yours, with individual Assert statements.